src/t_string.c

```diff
@@ -147,6 +130,34 @@ void setGenericCommand(client *c, int flags, robj *key, robj *val, robj *expire,
     }
 }
 
+/*
+ * Extract the `expire` argument of a given GET/SET command as an absolute timestamp in milliseconds.
+ *
+ * "client" is the client that sent the `expire` argument.
+ * "expire" is the `expire` argument to be extracted.
+ * "flags" represents the behavior of the command (e.g. PX or EX).
+ * "unit" is the original unit of the given `expire` argument (e.g. UNIT_SECONDS).
+ * "milliseconds" is output argument.
+ *
+ * If return C_OK, "milliseconds" output argument will be set to the resulting absolute timestamp.
+ * If return C_ERR, an error reply has been added to the given client.
+ */
+static int getExpireMillisecondsOrReply(client *c, robj *expire, int flags, int unit, long long *milliseconds) {
+    int ret = getLongLongFromObjectOrReply(c, expire, milliseconds, NULL);
+    if (ret != C_OK) {
+        return ret;
+    }
+    if (*milliseconds <= 0) {
+	addReplyErrorFormat(c,"invalid expire time in %s",c->cmd->name);
+	return C_ERR;
+    }
+    if (unit == UNIT_SECONDS) *milliseconds *= 1000;
```

Comment:
I think there was some discussion that expire could overflow during this multiplication. Might want to check it's still positive at the end.

Reply:
Good point. Done.